File: fibonacci/java_fibonacci/Fibonacci.java
public class Fibonacci {
  public static int fibonacci( int n ) {
    if( n == 0 ) {
      return 0;
    } else if( n == 1 ) {
      return 1;
    } else {
      return(fibonacci( n - 1 ) + fibonacci( n - 2 ));
    }
  }

  public static void main( String[] args ) {
    if( args.length != 0 ) {
      System.err.println( "Usage: java -jar Fibonacci.jar <noargs>");
      System.exit(1);
    }

    //System.out.println(fibonacci(new Integer(args[0])));
    for( int idx_i = 1; idx_i < 25; idx_i++ ) {
      System.out.println("fibonacci(" + idx_i + ")=<" + fibonacci(idx_i) + ">");
    }
  }
}

File: fibonacci/objc_fibonacci/fibonacci.m
#import <Foundation/Foundation.h>
#include <stdbool.h>
//comment
#define NSLog(FORMAT, ...) printf("%s\n", [[NSString stringWithFormat:FORMAT, ##__VA_ARGS__] UTF8String]);

@interface Fibonacci : NSObject
-(int)fibonacci:(int)num;
@end

@implementation Fibonacci
-(int)fibonacci:(int)num {
  if (num == 0) {
    return 0;
  }
  if (num == 1) {
    return 1;
  }
  return [self fibonacci:num - 1] + [self fibonacci:num - 2];
}
@end

int main (int argc, const char * argv[]) {
  NSAutoreleasePool * pool = [[NSAutoreleasePool alloc] init];
  if( argc != 1 ) {
    NSLog(@"Usage: %s <noargs>", argv[0]);
    exit(1);
  }
  int idx_i;

  id fibonacci = [[Fibonacci alloc] init];
  for( idx_i = 1; idx_i < 25; idx_i++ ) {
    NSLog(@"fibonacci(%d)=<%d>", idx_i, [fibonacci fibonacci:idx_i]);
  }
  [pool drain];
  return 0;
}
